### DISCUSSION TITLE
Can we prevent default values from being inserted when `null=False`? and give ValueError?

### DISCUSSION THREAD

**Discussion by devsarvesh92** [piccolo-orm/piccolo#1331] [Ideas]

**Comment by dantownsend** [comment]
There's a `required=True` arg you can specify for columns.

https://piccolo-orm.readthedocs.io/en/latest/piccolo/serialization/index.html#required-fields

This is currently just used in Pydantic models though, so you'd have use `create_pydantic` model to create a model, and use that for validation.

We might be able to do some validation at the `Table` level, but would need to think through that.

**Comment by devsarvesh92** [comment]
When we use `UUID(null=False, required=True, index=True)` in mixins, instead of throwing an error, it assigns a default value as a random UUID. Is this acceptable?
